The "PublishDefinitions" task failed unexpectedly in 1.5.0

I have a project that was using 1.4.12 however when I update to the latest 1.5.0 and build the exact the project I get the following error:

```
MSBuild version 17.4.1+fedecea9d for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
  PipelineProject -> C:\src\foo\bar\PipelineProject\PipelineProject\bin\Debug\net7.0\PipelineProject.dll
  Publishing all definitions inside bin\Debug\net7.0\PipelineProject.dll
  Using DefaultSharplinerConfiguration for configuration
  CiCdPipeline / pr.yml:
    Validating definition..
C:\Users\mathiasi\.nuget\packages\sharpliner\1.5.0\build\Sharpliner.targets(20,5): error MSB4018: The "PublishDefinitions" task failed unexpectedly. [C:\src\foo\bar\PipelineProject\PipelineProject\PipelineProject.csproj]
C:\Users\mathiasi\.nuget\packages\sharpliner\1.5.0\build\Sharpliner.targets(20,5): error MSB4018: (Line: 1, Col: 1, Idx: 0) - (Line: 1, Col: 1, Idx: 0): Expected SCALAR, SEQUENCE-START, MAPPING-START, or ALIAS, got MappingEnd [C:\src\foo\bar\PipelineProject\PipelineProject\PipelineProject.csproj]

Build FAILED.

C:\Users\mathiasi\.nuget\packages\sharpliner\1.5.0\build\Sharpliner.targets(20,5): error MSB4018: The "PublishDefinitions" task failed unexpectedly. [C:\src\foo\bar\PipelineProject\PipelineProject\PipelineProject.csproj]
C:\Users\mathiasi\.nuget\packages\sharpliner\1.5.0\build\Sharpliner.targets(20,5): error MSB4018: (Line: 1, Col: 1, Idx: 0) - (Line: 1, Col: 1, Idx: 0): Expected SCALAR, SEQUENCE-START, MAPPING-START, or ALIAS, got MappingEnd [C:\src\foo\bar\PipelineProject\PipelineProject\PipelineProject.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.80
```

--- a/src/Sharpliner/AzureDevOps/AzureDevOpsDefinition.cs
+++ b/src/Sharpliner/AzureDevOps/AzureDevOpsDefinition.cs
@@ -276,11 +276,16 @@ public abstract class AzureDevOpsDefinition
     /// Creates a publish task.
     /// </summary>
     protected static PublishTask Publish(string artifactName, string filePath, string? displayName = null)
-        => new PublishTask(filePath) with
+    {
+        var task = new PublishTask(filePath, artifactName);
+
+        if (displayName != null)
         {
-            DisplayName = displayName!,
-            Artifact = artifactName!,
-        };
+            task = task with { DisplayName = displayName };
+        }
+
+        return task;
+    }
 
     /// <summary>
     /// Creates a checkout task.

--- a/tests/Sharpliner.Tests/AzureDevOps/Model/Tasks/TaskBuilderTests.cs
+++ b/tests/Sharpliner.Tests/AzureDevOps/Model/Tasks/TaskBuilderTests.cs
@@ -197,6 +197,8 @@ public class TaskBuilderTests
                             ContinueOnError = false,
                             ArtifactType = ArtifactType.Pipeline,
                         },
+
+                        Publish("artifactName", "some/file/path.txt"),
                     }
                 }
             }
@@ -217,6 +219,9 @@ public class TaskBuilderTests
             displayName: Publish artifact
             artifact: Binary
             continueOnError: false
+
+          - publish: some/file/path.txt
+            artifact: artifactName
         """);
     }
 

